prboom2/src/dsda/hud_components/big_ammo.c

```diff
@@ -42,9 +45,24 @@ static void dsda_DrawComponent(void) {
     return;
 
   ammo = player->ammo[ammo_type];
+  max_ammo = player->maxammo[weaponinfo[player->readyweapon].ammo];
```

Comment:
We can just get the ammo percent directly and use it below, `ammo_percent = ammo * 100 / player->maxammo[ammo_type]`